cpp/src/arrow/compute/kernels/aggregate_basic_internal.h

```diff
@@ -256,8 +268,10 @@ struct MinMaxImpl : public ScalarAggregator {
     using ScalarType = typename TypeTraits<ArrowType>::ScalarType;
 
     std::vector<std::shared_ptr<Scalar>> values;
-    if (!state.has_values ||
-        (state.has_nulls && options.null_handling == MinMaxOptions::EMIT_NULL)) {
+    if (!state.has_values && options.skip_nulls && options.min_count == 0) {
+      values = {std::make_shared<ScalarType>(state.min),
+                std::make_shared<ScalarType>(state.max)};
```

Comment:
Code in this `if` clause looks not necessary as they are exactly the same as code in the last `else`.

Reply:
Indeed! :)
I've made this into a single if/else now.